locals.tf

```
@@ -1,7 +1,12 @@
 locals {
-  flattened_workspaces = merge(
-    [for org, workspaces in var.workspaces : { for ws in workspaces : "${org}-${ws}" => { org = org, ws = ws } }]...
-  )
+  flattened_workspaces = merge(flatten([for org, project in var.workspaces :
+    [for proj, workspace in project : { for ws in workspace : "${org}-${ws}" => {
```

To support same workspace name/patterns from different projects ? 

```suggestion
    [for proj, workspace in project : { for ws in workspace : "${org}-${proj}-${ws}" => {
```

---

hi @smoneyan , since this would change the value of  "workload_identity_pool_provider_id" . I'm not sure of it's reference anywhere else in the existing setups.